### ISSUE TITLE
Disco not connecting...

### ISSUE THREAD

**Issue by steelnick** [uavpal/disco4g#99]
Hey guys,
for several days I have been trying to soft mode my disco, but it is not working. I have huaweii e3372h-153 (written on the modem), but apparently working as a stick-mode (firmware 21.x)
In the options menu I saw the apn, no passwords or pins required. On a pc its is working, but not on the parrot. I have downgraded to firware version 1.7.0, but there is no difference. 
I have a question - in telnet mode, when I copy all the lines into the command prompt, they are starting to execute, and finally I have to press enter to reboot, when do I have to turn of the skycontroller? Before hitting enter, while the disco is rebooting, or after the reboot?

Thx in advance!

Nick

**Comment by ClassAxion** [comment]
"h" and "s" suffix is related to hardware version of the modem, not hilink / stick firmware.

**Comment by s0ftice** [comment]
it doesn't matter at what time you turn on the skycontroller.
when you say in the options menu on the PC you can see the APN, what software is that? Is it "Mobile Partner"?

you can try to post some debug output in here, see https://uavpal.com/disco/faq#debug

**Comment by steelnick** [comment]
Yes, its a software from the mobile partner (see the pic below)

![1](https://user-images.githubusercontent.com/90459487/133043962-872b1f43-4b55-4de2-b58e-d471a18eaa4e.png)

When I run ulogcat, I am getting this. Unfortunately I cannot translate it :D (sorry, I am a monkey)
![2](https://user-images.githubusercontent.com/90459487/133044022-465c7f2d-4ec1-49d6-9c46-ecb20f8fd27b.png)

**Comment by s0ftice** [comment]
1. Have you set the APN config file of the softmod to the same value? (internet.vivacom.bg)?
2. Do you know if the Username/password is mandatory by your carrier?
3. Not just `ulogcat`, please run `ulogcat |grep uavpal` as stated in the FAQ article I linked earlier.

**Comment by steelnick** [comment]
1. Yes, I had
2. there is such option, but it is not activated
3. It didn't work the first time I tried, now I did it again:
![3](https://user-images.githubusercontent.com/90459487/133076323-20015536-8e13-4d74-952a-675c27cdb638.png)

**Comment by steelnick** [comment]
One more thing - I have tried to create debug file, but obviously something goes wrong.
How am I supposed to create empty file, do I use notepad and then upload it by filezilla?

**Comment by s0ftice** [comment]
no ignore the debug log file.

2. it looks activated on your screenshot. pls check with your carrier
3. that's still without `| grep uavpal` ...

**Comment by steelnick** [comment]
2. ОК
3. hmm, I copied it from Your message, will try again...
When I hit enter, nothing happens
![4](https://user-images.githubusercontent.com/90459487/133086898-b22033f2-3b80-4b75-92c9-ebcee942b144.png)

**Comment by steelnick** [comment]
Nothing happens :(
Yes, the modem has led, when plugged in at first its off, after 3-5 sec it blinks with green couple of times, then cyan for 2 sec

**Comment by steelnick** [comment]
thinking about updating to latest firmware

**Comment by steelnick** [comment]
I installed the latest firmware and tried again... This time on ulogcat only I got this
![5](https://user-images.githubusercontent.com/90459487/133227497-fe3e8026-ed61-46b6-8271-407f2b4ec923.png)
Nothing else changed

**Comment by s0ftice** [comment]
this is the entire extract, which does not help... without `| grep uavpal` please ...
and if that's still empty 30 seconds after plugging in the modem, you probably did not install the mod correctly.

Are you coping the command lines one by one or as one big block?

**Comment by steelnick** [comment]
I reinstalled it, as follows:
1. telnet 192.168.42.1, enter
2. mv /data/ftp/internal_000/disco4g-* /tmp/disco4g, enter, gives new line and "/ #"
3. chmod +x /tmp/disco4g/*/*_install.sh, enter, gives new line and "/ #"
4. /tmp/disco4g/disco/disco_install.sh, enter, starts installing, at the end gives new line and "/ #"
5. /tmp/disco4g/skycontroller2/skycontroller2_install.sh, starts installing, at first is trying to connect, not connecting, tries again, connected, installing, at the end gives new line and "/ #"
6. reboot, enter
7. after reboot is done (skycontroller led is green) I am shutting off the controler. USB modem is flashing cyan every 2 sec, connecting, telnet ulogcat | grep uavpal - nothing :( 
Sorry for being a monkey, but - on windows "|" is one symbol, but in command prompt is like ":", I have tried with copying and the symbol changes.

**Comment by steelnick** [comment]
Here are some pictures from install:
![6](https://user-images.githubusercontent.com/90459487/133431300-b2040b9a-3e0c-48de-bddd-f6bfeaca4ebe.png)
![7](https://user-images.githubusercontent.com/90459487/133431315-bfb67f62-5292-40f8-b883-acc29263dadb.png)

**Comment by s0ftice** [comment]
Installation looks fine. The pipe symbol looks like a colon in Windows, that's also fine. Are you plugging the modem into the right or left USB port (looking from behind at CHUCK)? should be right. what output does `lsusb` show? (run in a telnet session about 10 second after the modem is plugged in)

What could be: your USB cable is a power cable only without data wires.

**Comment by steelnick** [comment]
Everything is as You have written, the cable has to be OTG, haven't tried it though.
1susb showed:
![8](https://user-images.githubusercontent.com/90459487/133562076-103a917d-eec5-4bd0-954e-8e11a25b0116.png)

**Comment by s0ftice** [comment]
I'm not at home currently to compare, can you run `lsusb` when the modem is not plugged in? Does it show the same 4 devices?

**Comment by steelnick** [comment]
Nope
![9](https://user-images.githubusercontent.com/90459487/133566158-b13f0ee8-ea04-490d-81a6-ccc2bd0131ad.png)

**Comment by s0ftice** [comment]
lsusb, not 1usb!

**Comment by steelnick** [comment]
Sorry, my bad! The devil is in the details! :)
Looks pretty much the same :(
![10](https://user-images.githubusercontent.com/90459487/133567772-0f7628de-a8c4-4c85-ade7-350ef3ef512d.png)
So, the problem is with the usb, too bad I have payed 4 euros for that "OTG" cable, will try to solder one tomorrow, any recommendations?

**Comment by s0ftice** [comment]
Yes, simple answer: your cable is either broken or does not have data wires at all. Try a different cable.

**Comment by steelnick** [comment]
I just soldered a new OTG cable, but I am getting the same answer. Is it possible the usb port of CHUCK to be the problem? Should I open it to check it, or there is another way?

**Comment by s0ftice** [comment]
you can try the other port, check https://uavpal.com/disco/wiki/How-to-fix-a-broken-Micro-USB-port-on-CHUCK (Option 2)
but you also have to make a special cable.

**Comment by steelnick** [comment]
OOOOK! Finally worked! After the start I got almost right away steady cyan :) 
Unfortunately now I have another problem - not connecting to zerotier, on the members tab shows no devices has joined

**Comment by steelnick** [comment]
I uninstalled the mod, installed it again, nothing changed :( am I loosing something on the way? On skycontroller everything worked fine - connected to zerotier. What could be the problem with the disco?
s0ftice, thank You for the help! I really appreciate it!

**Comment by steelnick** [comment]
I tried to reinstall the mod only with zerotier id, same problem - controller is connecting but the disco is not.
Ran the ulogcat|grep uavpal:
![11](https://user-images.githubusercontent.com/90459487/133999518-e06988bc-eef3-4391-80cb-7bb34292259a.png)

**Comment by s0ftice** [comment]
the modem seem connecting and disconnecting non-stop. Maybe the cable has a loose connection?

**Comment by steelnick** [comment]
the modem light is not blinking, may be is not a power issue. Will double check! Thx again :)

**Comment by steelnick** [comment]
the OTG shortening seemed a bit loose, I rewelded it.
![12](https://user-images.githubusercontent.com/90459487/134008698-0473575a-3365-404c-86bc-2c97058e050a.png)

**Comment by s0ftice** [comment]
I'm not sure. You could try to flash a Hilink firmware to the modem.
